gqcp/include/Basis/Transformations/SimpleOrbitalRotationGenerators.hpp

```diff
@@ -0,0 +1,149 @@
+// This file is part of GQCG-GQCP.
+//
+// Copyright (C) 2017-2020  the GQCG developers
+//
+// GQCG-GQCP is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// GQCG-GQCP is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with GQCG-GQCP.  If not, see <http://www.gnu.org/licenses/>.
+
+#pragma once
+
+
+#include "Mathematical/Representation/SquareMatrix.hpp"
+
+
+namespace GQCP {
+
+
+/**
+ *  MARK: SimpleOrbitalRotationGenerator implementation
+ */
+
+
+/**
+ *  An orbitalRotationGenerator that can be represented by a single vector.
+ *   
+ *  This class is used as a base class for `ROrbitalRotationGenerator` and `GOrbitalRotationGenerator`, since they are both expressed using a single vector, as opposed to `UOrbitalRotationGenerator`, which uses separate kappa vectors for alpha- and beta- generators. The word 'simple' is used here as an antonym for 'compound'.
+ * 
+ *  @tparam _Scalar                                           The scalar type used for a orbital rotation generator vector: real or complex.
+ *  @tparam _DerivedOrbitalRotationGenerator                  The type of the orbital rotation generator that derives from this class, enabling CRTP and compile-time polymorphism.
+ */
+template <typename _Scalar, typename _DerivedOrbitalRotationGenerators>
+class SimpleOrbitalRotationGenerators {
+
+public:
+    // The scalar type used for a orbital rotation generator vector: real or complex.
+    using Scalar = _Scalar;
+
+    // The type of the orbital rotation generator that derives from this class, enabling CRTP and compile-time polymorphism.
+    using DerivedOrbitalRotationGenerators = _DerivedOrbitalRotationGenerators;
+
+private:
+    // The number of spatial orbitals that can be rotated using these orbital rotation generators.
+    size_t number_of_spatial_orbitals;
+
+    // The strict upper/lower triangle of the kappa matrix.
+    VectorX<Scalar> kappa_vector;
+
+
+public:
+    /**
+     *  MARK: Constructors
+     */
+
+    /**
+     *  Create a `SimpleOrbitalRotationGenerator' from a given kappa vector.
```

Comment:
You're going to have to think a little more on how you define the 'kappa vector' in your documentation. For me and you, this is probably clear. But will it be clear for someone reading this documentation in 5 years?